sync2/interface.go

```diff
@@ -0,0 +1,36 @@
+package sync2
+
+import (
+	"context"
+
+	"github.com/libp2p/go-libp2p/core/host"
+
+	"github.com/spacemeshos/go-spacemesh/common/types"
+	"github.com/spacemeshos/go-spacemesh/fetch/peers"
+	"github.com/spacemeshos/go-spacemesh/p2p"
+	"github.com/spacemeshos/go-spacemesh/system"
+)
+
+//go:generate mockgen -typed -package=sync2_test -destination=./mocks_test.go -source=./interface.go
+
+type Fetcher interface {
+	system.AtxFetcher
+	Host() host.Host
+	Peers() *peers.Peers
+	RegisterPeerHash(peer p2p.Peer, hash types.Hash32)
+}
```

Comment:
I think you can just inline `system.AtxFetcher`:

```suggestion
type Fetcher interface {
	GetAtxs(context.Context, []types.ATXID, ...GetAtxOpt) error
	Host() host.Host
	Peers() *peers.Peers
	RegisterPeerHash(peer p2p.Peer, hash types.Hash32)
}
```

Interfaces should be defined on the user side not in a `system` package.

Reply:
With my other suggestions this can be further simplified to

```go
type Fetcher interface {
	GetAtxs(context.Context, []types.ATXID, ...system.GetAtxOpt) error
	RegisterPeerHashes(peer p2p.Peer, hash []types.Hash32)
}
```

Reply:
Fixed by merging your branch